guide/v3-upgrade.md

```diff
@@ -0,0 +1,56 @@
+# Upgrading to TDS V3
+
+As of **September 1st, 2019** TDS will move feature support to version 3 of the design system. This is being done primarily to facilitate the use of Styled Components V4. For a detailed view of our sunsetting schedule, please read our [roadmap page](roadmap.md). Additionally, more information on this subject can be found on the the [Reference Architecture CSS Document](https://github.com/telus/reference-architecture/blob/bb7059d135574c380d2865aa1bbdd633c2345461/development/css.md).
+
+## Requirements
+
+To upgrade to TDS V3, your repo will need **Styled Components 4.1** and **React 16.8** installed. These packages are peer dependencies of all TDS V3 components, and lower versions will cause incompatibilities.
+
+### Upgrading from Styled Components V3 to V4
+
+The Styled Components team has provided a codemod to make the upgrade process from V3 to V4 as fast as possible. It can be accessed [here](https://github.com/styled-components/styled-components-codemods). If you would like more information on the changes and improvements made from V3 to V4, check out the [Styled Components V4 announcement](https://medium.com/styled-components/announcing-styled-components-v4-better-faster-stronger-3fe1aba1a112).
+
+### Upgrading from React 15 to React 16
+
+Upgrading from React 15 to React 16 could be as straightforward as upgrading the dependency without any code changes. As noted on the [React 16 announcement post](https://reactjs.org/blog/2017/09/26/react-v16.0.html#upgrading):
+
+> With minor exceptions, if your app runs in 15.6 without any warnings, it should work in 16.
+
+So, if your app does not currently display any warnings from React, the upgrade could take almost no time at all! If your app does display warnings however, there are [official codemods](https://github.com/reactjs/react-codemod) that may be able to expedite the process of upgrading.
+
+## Developing in Styled components
```

Comment:
Just a though, since Styled Components maintainers refers to the product always as styled-components, should we do that as well?

Product reference: **styled-components**
A component (or more) that is styled: **styled component(s)**

Reply:
Should be July 1st